cou/commands.py

```diff
@@ -124,6 +124,19 @@ def get_subcommand_common_opts_parser() -> argparse.ArgumentParser:
         action=argparse.BooleanOptionalAction,
         default=argparse.SUPPRESS,
     )
+    subcommand_common_opts_parser.add_argument(
+        "--archive",
+        help="Archive old database data (nova) before cloud upgrade.\n"
+        "Default to enabling archive.",
+        action=argparse.BooleanOptionalAction,
+        default=argparse.SUPPRESS,
+    )
+    subcommand_common_opts_parser.add_argument(
+        "--archive-batch-size",
+        help="Batch size for nova old database data archiving.\n" "Default is 1000.",
+        type=int,
```

Comment:
It would be nice to ensure that this is a positive integer and we can do that by having a callable function on type:

```python
def positive_batch_size(value: str) -> int:
    ivalue = int(value)
    if ivalue <= 0:
        raise argparse.ArgumentTypeError("Batch size should be a positive integer")
    return ivalue
```

Then we can add `type=positive_batch_size`